### PR TITLE
UI Improvements for materials

### PR DESCRIPTION
**Pull request by allenwq**
screen shot 2015-10-23 at 4 15 10 pm

1. Adopt settings to materials title 
2. Using breadcrumbs to show the folder hierarchy
3. Other UI improvements

### CODE REVIEW

File: app/controllers/course/material/controller.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  load_and_authorize_resource :folder, through: :course, through_association: :material_folders,
+                                       class: Course::Material::Folder.name
+
+  before_action :check_component
+  before_action :load_settings
+  before_action :add_folder_breadcrumb
+
+  private
+
+  # @return [Course::MaterialsComponent | nil] The materials component or nil if disabled.
```
Comment (lowjoel) [inline app/controllers/course/material/controller.rb]: I usually do not put spaces before and after |

File: app/helpers/course/material/folders_helper.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module Course::Material::FoldersHelper
+  def display_name(folder, settings)
+    folder.root? ? settings.title || t('course.material.sidebar_title') : folder.name
```
Comment (lowjoel) [inline app/helpers/course/material/folders_helper.rb]: settings is accessible from controller.settings right? Or declare settings to be `helper_method`

File: app/controllers/course/material/controller.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  before_action :check_component
+  before_action :add_folder_breadcrumb
+
+  private
+
+  # @return [Course::MaterialsComponent|nil] The materials component or nil if disabled.
+  def component
+    current_component_host[:course_materials_component]
+  end
+  helper_method :component
```
Comment (allenwq) [inline app/controllers/course/material/controller.rb]: @lowjoel How about it now ? I have made this a helper_method, and we can get settings through `component.settings`

Comment (lowjoel) [inline app/controllers/course/material/controller.rb]: great. thanks.

File: app/views/course/material/folders/show.html.slim [hunk 4]
```diff
@@ -1,4 +1,4 @@
-= page_header(@folder.name)
+= page_header(display_name(@folder))
```
Comment (lowjoel) [inline app/views/course/material/folders/show.html.slim]: we have other helpers like this, right? is this a consistent name?

Comment (allenwq) [inline app/views/course/material/folders/show.html.slim]: We have a `display_user` ?  How about `folder_display_name` ?

Comment (lowjoel) [inline app/views/course/material/folders/show.html.slim]: grep the source. if this is local, I think it's fine.

Comment (lowjoel) [inline app/views/course/material/folders/show.html.slim]: but very likely you may need to have a global helper, because you want to link to it.

Comment (allenwq) [inline app/views/course/material/folders/show.html.slim]: I greped, this seems to be the only occurrence. and sorry, why I want to link to the folder ?

Comment (lowjoel) [inline app/views/course/material/folders/show.html.slim]: link to all files? I don't know.

The name of the function is called `display_user` and `link_to_user`

Comment (allenwq) [inline app/views/course/material/folders/show.html.slim]: OK, Will probably need a helper like `display_files(folder)` and also a view partial for it. I'll do it in future PRs. 

So do you want me to rename this function  ?

Comment (lowjoel) [inline app/views/course/material/folders/show.html.slim]: Display folder and link to folder? No need to list the folder contents.

**Comment by allenwq** [comment]
@lowjoel Updated.

**Comment by allenwq** [comment]
@lowjoel Thanks !

**Pull request merged**